File: lzt_utils/Lorenzetti/CaloCluster.cpp
#include <vector>
#include "Lorenzetti/CaloCluster.h"
#include "Lorenzetti/Event.h"
#include "Lorenzetti/CaloCell.h"
#include "Lorenzetti/xAOD.h"
#include <ROOT/RVec.hxx>

namespace Lorenzetti {

    CaloCluster_t makeCaloCluster(ROOT::RVec<xAOD::EventSeed_t> &event_seeds, ROOT::RVec<xAOD::CaloCell_t> &calo_cells){
        CaloCluster_t calo_cluster = {
            .e=0,
            .et=0,
            .eta=0,
            .phi=0,
            .deta=0,
            .dphi=0,
            .e0=0,
            .e1=0,
            .e2=0,
            .e3=0,
            .ehad1=0,
            .ehad2=0,
            .ehad3=0,
            .etot=0,
            .e233=0,
            .e237=0,
            .e277=0,
            .emaxs1=0,
            .emaxs2=0,
            .e2tsts1=0,
            .reta=0,
            .rphi=0,
            .rhad=0,
            .rhad1=0,
            .eratio=0,
            .f0=0,
            .f1=0,
            .f2=0,
            .f3=0,
            .weta2=0,
            .secondR=0,
            .lambdaCenter=0,
            .secondLambda=0,
            .fracMax=0,
            .lateralMom=0,
            .longitudinalMom=0
        };
        return calo_cluster;
    }


}

File: lzt_utils/calo_cluster.py
from typing import TypedDict, Tuple


class CaloCluster(TypedDict):
    e: float
    et: float
    eta: float
    phi: float
    deta: float
    dphi: float
    e0: float
    e1: float
    e2: float
    e3: float
    ehad1: float
    ehad2: float
    ehad3: float
    etot: float
    e233: float
    e237: float
    e277: float
    emaxs1: float
    emaxs2: float
    e2tsts1: float
    reta: float
    rphi: float
    rhad: float
    rhad1: float
    eratio: float
    f0: float
    f1: float
    f2: float
    f3: float
    weta2: float
    secondR: float
    lambdaCenter: float
    secondLambda: float
    fracMax: float
    lateralMom: float
    longitudinalMom: float


def get_eta_phi_range(cluster: CaloCluster) -> Tuple[Tuple[float, float],
                                                     Tuple[float, float]]:
    """
    Get the eta and phi range of a cluster.

    Parameters
    ----------
    cluster : CaloCluster
        The cluster.

    Returns
    -------
    Tuple[Tuple[float, float], Tuple[float, float]]
        The eta and phi range of the cluster.
    """
    eta_range = (
        cluster.eta - cluster.deta,
        cluster.eta + cluster.deta
    )
    phi_range = (
        cluster.phi - cluster.dphi,
        cluster.phi + cluster.dphi
    )
    return eta_range, phi_range
